modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotRestoreProcess.java

```diff
@@ -1395,7 +1395,7 @@ private void finishCacheStop(UUID reqId, Map<UUID, Boolean> res, Map<UUID, Throw
      * @return Result future.
      */
     private IgniteInternalFuture<Boolean> incrementalSnapshotRestore(UUID reqId) {
-        if (ctx.clientNode())
+        if (ctx.clientNode() || !CU.baselineNode(ctx.discovery().localNode(), ctx.state().clusterState()))
```

Comment:
Nice catch, thanks!

I think we should use another check here. Let's check local node is included into `opCtx0.nodes()`. This collection bases on pre-defined discoCache, so it looks like it's more reliable way to verify node. WDYT?

Also, I think we should check `opCtx0 != null` also.

Reply:
Thanks, fixed.

Reply:
> we should check opCtx0 != null also.

Added.